src/flint/fmpz.jl

```diff
@@ -627,7 +627,23 @@ function ^(x::fmpz, y::Int)
       deepcopy(x)
    else
       z = fmpz()
-      ccall((:fmpz_pow_ui, libflint), Nothing, (Ref{fmpz}, Ref{fmpz}, UInt), z, x, UInt(y))
+      ccall((:fmpz_pow_ui, libflint), Nothing,
+            (Ref{fmpz}, Ref{fmpz}, UInt), z, x, UInt(y))
+      z
+   end
+end
+
+function ^(x::fmpz, y::UInt)
```

Comment:
I see no reason why this cannot be combined into the union above. The compiler should know that the condition y<0 is always false.

Reply:
Done, thanks!